app/dashboard/browse/search/profiles/[id]/page.tsx

```diff
@@ -1,9 +1,65 @@
+"use client"
 import PageHeading from "@/components/dashboard/PageHeading";
+import { useToast } from "@/components/ui/use-toast";
+import { getSingleProfile } from "@/utils/scripts/search";
+import { useEffect, useState } from "react";
 
 export default function ProfilePage({ params }: { params: { id: string } }) {
+
+  const { toast } = useToast();
+
+  const [errorFetching, setErrorFetching] = useState(false);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [bio, setBio] = useState('');
+  const [orgName, setOrgName] = useState('');
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const profileData = await getSingleProfile(params.id);
+      if (profileData === null) {
+        toast({
+          variant: "destructive",
+          title: "Unable to retrieve profile",
+          description: "Reload page or check another time",
+        });
+        setErrorFetching(true);
+        return;
+      }
+      setName(profileData.full_name);
+      setEmail(profileData.email);
+      setBio(profileData.bio);
+      setOrgName(profileData.FK_organizations.name);
+
+    }
+
+    fetchData();
+  }, []);
+
+
   return (
     <div className="">
       <PageHeading header="Search" description="Public profile" />
+      {errorFetching ? (
+        <div className="mt-8 text-center">
+          There was an error retrieving profile. Reload or try again later.
+        </div>
+      ) : (
+        <div className="flex mt-8">
+          <div className="2xl:w-[50vw] w-[50vw] 2xl:h-[65vh] h-[55vh] 2xl:px-10 px-6 2xl:py-8 py-6 bg-[url('/assets/backgrounds/section_background.svg')] bg-no-repeat bg-cover bg-bottom rounded-3xl shadow-inner shadow-xl">
```

Comment:
<img width="1506" alt="Screenshot 2024-04-28 at 7 35 38 PM" src="https://github.com/nam-t24/Senior-CS-Project/assets/71054634/7b54626d-bf05-4e76-b7d6-cdb277beb124">
The text are a little too big for smaller screens

Reply:
same with organizations page